lib/pub_sub/railties/active_record.rb

```diff
@@ -5,9 +5,14 @@ def self.included(klass)
         klass.send :include, InstanceMethods
         klass.send :extend, ClassMethods
 
-        klass.after_save :publish_changes, if: -> { changed? }
-        klass.after_destroy :publish_changes
-        klass.after_touch :publish_changes
+        klass.after_save { @__pub_sub_record_unchanged = !changed? }
+        # Not everything executes the `after_save` callback (like touch and destroy),
+        # so use `unless` logic so the uninitialized @__xxx_unchanged is nil, which equals false.
+        klass.after_commit :publish_changes, unless: Proc.new {
+          # return current unchanged status and reset unchanged status to false
+          x, @__pub_sub_record_unchanged = @__pub_sub_record_unchanged, false
```

Comment:
Granted very concise 😄 , but this piece is a bit hard to follow. I'd lean toward being more explicit and clear as to when changes are published. This is what I have in mind as an example.

```ruby
#excplit cases that mark a record to be published
klass.after_save :__pubsub_mark_to_publish!, if: :changed?
klass.after_destroy :__pubsub_mark_to_publish!
klass.after_touch :__pubsub_mark_to_publish!
klass.after_commit :publish_changes, if: __pubsub_changes_to_publish?

def __pubsub_mark_to_publish!
  @__pubsub_changes_to_publish = true
end

def __pub_sub_marked_to_publish?
  @__pubsub_changes_to_publish
end

module InstanceMethods
  def publish_changes
    @__pubsub_changes_to_publish = false 
    # ...
  end
# ... 
```